lunatrace/bsl/backend/src/github/actions/generate-snapshots-for-repositories.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1 
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { GithubRepositoryInfo } from "../../types/github";
+import {log} from "../../utils/log";
+import {catchError, threwError} from "../../utils/try";
+import {generateGithubGraphqlClient} from "../api";
+import {getInstallationAccessToken} from "../auth";
+
+import {generateSnapshotForRepository} from "./generate-snapshot-for-repository";
+
+export async function generateSnapshotsForGithubRepos(installationId: number, githubRepos: GithubRepositoryInfo[]) {
+  const installationToken = await getInstallationAccessToken(installationId);
+
+  if (installationToken.error) {
+    log.error('unable to get installation token', {
+      error: installationToken.msg
+    });
+    return;
+  }
+
+  const github = generateGithubGraphqlClient(installationToken.res);
```

Comment:
We should just pass this Client in as an argument to this function. The fact that we have to call `getInstallationAccessToken` and then setup the client manually with the `installationToken` is brutal. We're already doing that in a bunch of places in our codebase.